gitlab_jobs.py

```diff
@@ -69,30 +70,39 @@ def main():
         template = "Last {n} successful pipelines of {project} {ref}:"
     print(template.format(
         n=args.limit, ref=args.branch, project=project.name))
-    pipelines = project.pipelines.list(
-        scope='finished', status='success', ref=args.branch,
-        per_page=args.limit)
-    for pipeline in pipelines:
-        template = "  {id} (commit {sha}"
-        if args.verbose:
-            template += " by {user[name]}"
-        if args.branch is None:
-            template += " on {ref}"
-        template += ", duration {duration_min:.1f}m)"
-        # pipeline data returned in the list contains only a small subset
-        # of information, so we need an extra HTTP GET to fetch duration
-        # and user
-        pipeline = project.pipelines.get(pipeline.id)
-        pipeline_durations.append(pipeline.duration)
-        duration_min = pipeline.duration / 60.0
-        print(template.format(
-            duration_min=duration_min, **pipeline.attributes))
-        for job in pipeline.jobs.list(scope='success', all=True):
-            job_durations[job.name].append(job.duration)
+
+    max_per_page = 100
+    pages = math.ceil(args.limit / max_per_page)
+    for page in range(1, pages+1):
+        per_page = max_per_page
+        last_page_leftover = args.limit % max_per_page
+        if page == pages and last_page_leftover != 0:
+            per_page = last_page_leftover
+
```

Comment:
The diff (and also the code itself, I think) can be made more readable if you extract this bit into a helper function.  Something like

```python
def get_pipelines(args):
    ...
    for page in ...
        ...
        for pipeline in project.pipelines.list(...):
            yield pipeline
```

and then in main() you'd replace the `pipelines = ...` assignment to call the new helper and then you can keep the same `for pipeline in pipelines:` loop at the same indentation.